Tests/SuperLinq.Async.Test/HasDuplicatesTest.cs

```diff
@@ -0,0 +1,214 @@
+﻿namespace Test.Async;
+
+public class HasDuplicatesTest
+{
+	[Fact]
+	public async Task When_Asking_For_Duplicates_On_Sequence_Without_Duplicates_Then_False_Is_Returned()
+	{
+		await using var asyncSequence = TestingSequence.Of(
+			"FirstElement",
+			"SecondElement",
+			"ThirdElement");
+
+		var hasDuplicates = await asyncSequence.HasDuplicates();
+
+		Assert.False(hasDuplicates);
+	}
+
+	[Fact]
+	public async Task When_Asking_For_Duplicates_On_Sequence_Projection_Without_Duplicates_Then_False_Is_Returned()
+	{
+		await using var asyncSequence = TestingSequence.Of(
+			new DummyClass("FirstElement"),
+			new DummyClass("SecondElement"),
+			new DummyClass("ThirdElement"));
+
+		var hasDuplicates = await asyncSequence.HasDuplicates(x => x.ComparableString);
+
+		Assert.False(hasDuplicates);
+	}
+
+	[Fact]
+	public async Task When_Asking_For_Duplicates_On_Sequence_With_Duplicates_Then_True_Is_Returned()
+	{
+		await using var asyncSequence = TestingSequence.Of(
+			"FirstElement",
+			"DUPLICATED_STRING",
+			"DUPLICATED_STRING",
+			"ThirdElement");
+
+		var hasDuplicates = await asyncSequence.HasDuplicates();
+
+		Assert.True(hasDuplicates);
+	}
+
+	[Fact]
+	public async Task When_Asking_For_Duplicates_On_Sequence_Projection_With_Duplicates_Then_True_Is_Returned()
+	{
+		await using var asyncSequence = TestingSequence.Of(
+			new DummyClass("FirstElement"),
+			new DummyClass("DUPLICATED_STRING"),
+			new DummyClass("DUPLICATED_STRING"),
+			new DummyClass("ThirdElement"));
+
+		var hasDuplicates = await asyncSequence.HasDuplicates(x => x.ComparableString);
+
+		Assert.True(hasDuplicates);
+	}
+
+	[Fact]
+	public async Task When_Asking_For_Duplicates_On_Sequence_Projection_With_Duplicates_Then_It_Does_Not_Iterate_Unnecessary_On_Elements()
+	{
+		await using var asyncSequence = TestingSequence.OfWithFailure(
+			new DummyClass("FirstElement"),
+			new DummyClass("DUPLICATED_STRING"),
+			new DummyClass("DUPLICATED_STRING"));
+
+		var record = await Record.ExceptionAsync(async () => await asyncSequence.HasDuplicates(x => x.ComparableString));
```

Comment:
Sorry, last comment, and we should be good.

What is this `Record.Exception()`? We don't use it anywhere else in the codebase.

If the operator is expected to throw an exception, we should use `Assert.Throws<>()`; if not, then there's no reason for the `Record`, because _any_ exception will cause a failed test. `Assert`ing that a `Record` recorded no exceptions doesn't add any value to the test, as far as I'm concerned?

Reply:
It's the equivalent of Assert.DoesNotThrow<> in nUnit.

but reading xUnit documentation, they say the same as you
![image](https://github.com/viceroypenguin/SuperLinq/assets/14810960/4eeaa201-0b08-402d-967d-01d756584abd),

i'll edit that!